superdesk/publish/formatters/ninjs_ftp_formatter.py

```diff
@@ -47,48 +42,212 @@ def _transform_to_ninjs(self, article, subscriber, recursive=True):
         :param recursive:
         :return:
         """
-
+        # Get the path that the renditions will be pushed to
+        self.path = subscriber.get("destinations")[0].get("config").get("associated_path")
         include_original = subscriber.get("destinations")[0].get("config").get("include_original", False)
         if include_original:
             self.internal_renditions = ["original"]
 
-        ninjs = super()._transform_to_ninjs(article, subscriber, recursive)
+        formatted_article = deepcopy(article)
 
-        # Get the path that the renditions will be pushed to
-        path = subscriber.get("destinations")[0].get("config").get("associated_path")
+        if article.get("type") == "text" and recursive:
+            self.apply_product_filtering_to_associations(formatted_article, subscriber)
+
+        ninjs = super()._transform_to_ninjs(formatted_article, subscriber, recursive)
+
+        renditions = ninjs.get("renditions")
+        if renditions:
+            for name, rendition in renditions.items():
+                rendition["href"] = (
+                    self.path.lstrip("/")
```

Comment:
Why remove the `/` prefix? It was there in the original implementation

Reply:
The idea is to make the paths relative rather than absolute on the FTP server

Reply:
@marwoodandrew An issue I have with the change to the FTP path, will this not require a change to the config after updating?

I understand your reasons, but we will have to make sure this is communicated as a breaking change when we update clients that use this formatter/transmitter, so that their configs are updated.

Reply:
I don't think anyone else is using ninjs ftp formatter, so as long as we add some notice we should be fine